### PR TITLE
No need to ScrubDNS anymore

### PR DESCRIPTION
**Pull request by wlan0**
@thockin This removes the DNS scrubber from kubelet.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36785)

### CODE REVIEW

**Comment by yujuhong** [comment]
Reassigned to @thockin to review or delegate.

File: pkg/kubelet/kubelet_network.go [hunk 1]
```diff
@@ -89,25 +89,15 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
...
-func (kl *Kubelet) parseResolvConf(reader io.Reader) (nameservers []string, searches []string, err error) {
-	var scrubber dnsScrubber
-	if kl.cloud != nil {
-		scrubber = kl.cloud
-	}
-	return parseResolvConf(reader, scrubber)
-}
-
 // A helper for testing.
 type dnsScrubber interface {
```
Comment (thockin) [inline pkg/kubelet/kubelet_network.go]: not needed any more?

File: pkg/kubelet/kubelet_network.go [hunk 2]
```diff
@@ -137,10 +127,6 @@ func parseResolvConf(reader io.Reader, dnsScrubber dnsScrubber) (nameservers []s
 		}
 	}
 
-	// Give the cloud-provider a chance to post-process DNS settings.
```
Comment (thockin) [inline pkg/kubelet/kubelet_network.go]: leave me a TODO or at least a comment saying that this used to scrub DNS for each cloud, but that doesn't make sense once cloud-provider is factorerd out.

Comment (wlan0) [inline pkg/kubelet/kubelet_network.go]: @thockin - Added the comment

**Comment by thockin** [comment]
Thanks!!!

**Comment by thockin** [comment]
@k8s-bot ok to test

**Comment by wlan0** [comment]
thanks @thockin :-)

**Comment by luxas** [comment]
Cool to see progress.
Please ping me as well @wlan0 :)

**Comment by wlan0** [comment]
@luxas Absolutely! i'll keep you updated on all future PRs :)

**Pull request merged**